src/Mvc/Mvc.ApiExplorer/test/DefaultApiDescriptionProviderTest.cs

```diff
@@ -243,6 +243,46 @@ public void GetApiDescription_CreatesMultipleDescriptionsForMultipleHttpMethods(
         }
     }
 
+    [Theory]
+    [InlineData("api/products/{id}", nameof(BindingSource.Path), true)]
+    [InlineData("api/products", nameof(BindingSource.Path), false)]
+    [InlineData("api/products", nameof(BindingSource.Body), true)]
+    [InlineData("api/products", nameof(BindingSource.Query), true)]
+    public void GetApiDescription_WithInferredBindingSource(
+        string template,
+        string inferredBindingSourceName,
+        bool inferredParameterShouldBeIncluded)
```

Comment:
Can we make these separate tests instead of ? `GetApiDescription_WithInferredBindingSource_IncludesPathParametersWhenPresentInRoute` / `GetApiDescription_WithInferredBindingSource_ExcludesPathParametersWhenNotPresentInRoute`?

Also do we need tests that verifies we don't change this if the BindingSource is not inferred?